clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -287,22 +338,29 @@ public CompletableFuture<Map<TopicIdPartition, Acknowledgements>> commitSync(
                         acknowledgementsMapForNode.put(tip, acknowledgements);
 
                         metricsManager.recordAcknowledgementSent(acknowledgements.size());
-                        log.debug("Added sync acknowledge request for partition {} to node {}", tip.topicPartition(), node.id());
+                        log.debug("Added sync acknowledge request for partition {} to node {}", tip.topicPartition(),
+                                node.id());
                         resultCount.incrementAndGet();
                     }
                 }
-                acknowledgeRequestStates.add(new AcknowledgeRequestState(logContext,
-                        ShareConsumeRequestManager.class.getSimpleName() + ":1",
-                        deadlineMs,
-                        retryBackoffMs,
-                        retryBackoffMaxMs,
-                        sessionHandler,
-                        nodeId,
-                        acknowledgementsMapForNode,
-                        this::handleShareAcknowledgeSuccess,
-                        this::handleShareAcknowledgeFailure,
-                        resultHandler
-                ));
+                acknowledgeRequestStates.putIfAbsent(nodeId, new Pair<>(null, null));
+
+                // There can only be one commitSync()/close() happening at a time. So per node,
```

Comment:
This is true but it would be simple to ensure that using a bit of defensive code. If we made an error in the code later, we might end up overwriting something we didn't expect and not notice.

Reply:
Yes makes sense, have added a check now.